apis/python/src/tiledbsoma/common.cc

```diff
@@ -135,7 +135,7 @@ tiledb_datatype_t np_to_tdb_dtype(py::dtype type) {
     if (kind.is(py::str("S")))
         return TILEDB_STRING_ASCII;
     if (kind.is(py::str("U")))
-        return TILEDB_STRING_UTF8;
+        return TILEDB_STRING_UTF32;
```

Comment:
This looks weird and it looks wrong -- even though it isn't -- please add a clear comment why we need to do this for multi-package API compliance.